modules/lang-painless/src/main/java/org/elasticsearch/painless/ir/ArgumentsNode.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.painless.ir;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class ArgumentsNode extends ExpressionNode {
+
+    /* ---- begin tree structure ---- */
+
+    private final List<ExpressionNode> argumentNodes = new ArrayList<>();
```

Comment:
These are the actual arguments, correct? If so, consider renaming to `arguments`.  It's a bit confusing to have an `ArgumentsNode` type with field `argumentNodes`.

Reply:
I'm open to suggestions here. I know the names are a bit confusing and clash, but they do both describe accurately what each item is. I have postfixed all the "ir" nodes with two types of data - tree structure and local data. All tree structure members end with `Node` to help differentiate this from typical data. This also creates a general consistency between all nodes.